src/coreclr/vm/dllimportcallback.cpp

```diff
@@ -1318,148 +517,10 @@ VOID UMThunkMarshInfo::RunTimeInit()
         pFinalILStub = JitILStub(pStubMD);
     }
 
-#if defined(TARGET_X86)
-    MetaSig sig(pMD);
-    int numRegistersUsed = 0;
-    UINT16 cbRetPop = 0;
-
-    //
-    // cbStackArgSize represents the number of arg bytes for the MANAGED signature
-    //
-    UINT32 cbStackArgSize = 0;
-
-    int offs = 0;
-
-#ifdef UNIX_X86_ABI
-    if (HasRetBuffArgUnmanagedFixup(&sig))
-    {
-        // callee should pop retbuf
-        numRegistersUsed += 1;
-        offs += STACK_ELEM_SIZE;
-        cbRetPop += STACK_ELEM_SIZE;
-    }
-#endif // UNIX_X86_ABI
-
-    for (UINT i = 0 ; i < sig.NumFixedArgs(); i++)
-    {
-        TypeHandle thValueType;
-        CorElementType type = sig.NextArgNormalized(&thValueType);
-        int cbSize = sig.GetElemSize(type, thValueType);
-        if (ArgIterator::IsArgumentInRegister(&numRegistersUsed, type, thValueType))
-        {
-            offs += STACK_ELEM_SIZE;
-        }
-        else
-        {
-            offs += StackElemSize(cbSize);
-            cbStackArgSize += StackElemSize(cbSize);
-        }
-    }
-    m_cbStackArgSize = cbStackArgSize;
-    m_cbActualArgSize = (pStubMD != NULL) ? pStubMD->AsDynamicMethodDesc()->GetNativeStackArgSize() : offs;
-
-    PInvokeStaticSigInfo sigInfo;
-    if (pMD != NULL)
-        new (&sigInfo) PInvokeStaticSigInfo(pMD);
-    else
-        new (&sigInfo) PInvokeStaticSigInfo(GetSignature(), GetModule());
-    if (sigInfo.GetCallConv() == pmCallConvCdecl)
-    {
-        m_cbRetPop = cbRetPop;
-    }
-    else
-    {
-        // For all the other calling convention except cdecl, callee pops the stack arguments
-        m_cbRetPop = cbRetPop + static_cast<UINT16>(m_cbActualArgSize);
-    }
-#endif // TARGET_X86
-
-#endif // TARGET_X86 && !FEATURE_STUBS_AS_IL
-
     // Must be the last thing we set!
     InterlockedCompareExchangeT<PCODE>(&m_pILStub, pFinalILStub, (PCODE)1);
 }
 
-#if defined(TARGET_X86) && defined(FEATURE_STUBS_AS_IL)
-VOID UMThunkMarshInfo::SetupArguments(char *pSrc, ArgumentRegisters *pArgRegs, char *pDst)
-{
-    MethodDesc *pMD = GetMethod();
-
-    _ASSERTE(pMD);
-
-    //
-    // x86 native uses the following stack layout:
-    // | saved eip |
-    // | --------- | <- CFA
-    // | stkarg 0  |
-    // | stkarg 1  |
-    // | ...       |
-    // | stkarg N  |
-    //
-    // x86 managed, however, uses a bit different stack layout:
-    // | saved eip |
-    // | --------- | <- CFA
-    // | stkarg M  | (NATIVE/MANAGE may have different number of stack arguments)
-    // | ...       |
-    // | stkarg 1  |
-    // | stkarg 0  |
-    //
-    // This stub bridges the gap between them.
-    //
-    char *pCurSrc = pSrc;
-    char *pCurDst = pDst + m_cbStackArgSize;
-
-    MetaSig sig(pMD);
-
-    int numRegistersUsed = 0;
-
-#ifdef UNIX_X86_ABI
-    if (HasRetBuffArgUnmanagedFixup(&sig))
```

Comment:
I am curious - where is this case handled after this change?

Reply:
This is handled in `Compiler::getReturnTypeForStruct` under the `#ifdef UNIX_X86_ABI` block after this change.

Reply:
Great, thanks for the info!